diff -dur -x '*.orig' jabberd-2.0s2.orig/tools/Makefile.am jabberd-2.0s2/tools/Makefile.am
--- jabberd-2.0s2.orig/tools/Makefile.am	2003-12-16 02:01:34.000000000 +0100
+++ jabberd-2.0s2/tools/Makefile.am	2004-02-10 13:36:11.462922856 +0100
@@ -3,6 +3,7 @@
 
 edit = sed \
 	-e 's,@sysconfdir\@,$(sysconfdir),g' \
+	-e 's,@libdir\@,$(libdir)/jabberd,g' \
 	-e 's,@VERSION\@,$(VERSION),g' \
        -e 's,@bindir\@,$(bindir),g'
 
