diff -ur jabberd-2.2.17.orig/tools/jabberd.in jabberd-2.2.17/tools/jabberd.in
--- jabberd-2.2.17.orig/tools/jabberd.in	2013-07-23 16:51:33.000000000 +0000
+++ jabberd-2.2.17/tools/jabberd.in	2013-07-23 16:52:28.974241647 +0000
@@ -14,6 +14,7 @@
 use IO::Select;
 use POSIX;
 use POSIX qw(setsid);
+use Proc::Daemon;
 
 
 #-----------------------------------------------------------------------------
@@ -47,7 +48,24 @@
     &usage() if (!defined($opts{c}) || ($opts{c} eq ""));
 }
 
-$debug = 1 if exists($opts{D});
+if (exists($opts{D})) {
+       $debug = 1
+}
+else {
+       Proc::Daemon::Init;
+       open(PID,">/var/run/jabberd.pid");
+       print PID "$$\n";
+       close(PID);
+}
+
+if ($< == 0) {
+        my ($uid, $gid) = (getpwnam("jabber"))[2,3];
+        $)="$gid $gid" or die "Cannot set EGID and groups";
+        $(=$gid or die "Cannot set GID";;
+        $>=$uid or die "Cannot set EUID";
+        $<=$> or die "Cannot set UID";;
+}
+
 $daemon = 1 if exists($opts{b});
 
 #-----------------------------------------------------------------------------
