--- jabberd-2.1.22/storage/authreg_mysql.c.orig	2008-01-21 16:16:58.000000000 +0100
+++ jabberd-2.1.22/storage/authreg_mysql.c	2008-01-22 14:57:56.000000000 +0100
@@ -421,8 +421,10 @@
 
     mysql_query(conn, "SET NAMES 'utf8'");
 
-    /* Set reconnect flag to 1 (set to 0 by default from mysql 5 on) */
-    conn->reconnect = 1;
+    {
+	my_bool reconnect = 1;
+	mysql_options(conn, MYSQL_OPT_RECONNECT, (char *)&reconnect);
+    }
 
     ar->user_exists = _ar_mysql_user_exists;
     ar->get_password = _ar_mysql_get_password;
--- jabberd-2.1.22/storage/storage_mysql.c.orig	2008-01-21 16:16:58.000000000 +0100
+++ jabberd-2.1.22/storage/storage_mysql.c	2008-01-22 14:57:49.000000000 +0100
@@ -632,8 +632,10 @@
         return st_FAILED;
     }
 
-    /* Set reconnect flag to 1 (set to 0 by default from mysql 5 on) */
-    conn->reconnect = 1;
+    {
+	my_bool reconnect = 1;
+	mysql_options(conn, MYSQL_OPT_RECONNECT, (char *)&reconnect);
+    }
 
     data = (drvdata_t) calloc(1, sizeof(struct drvdata_st));
 
