*** sx/io.c.orig        Mon Sep  6 22:08:27 2004
--- sx/io.c     Mon Sep  6 22:08:30 2004
***************
*** 107,111 ****

              /* hand it to the app */
!             if(plugin_error == 0)
                  _sx_event(s, event_PACKET, (void *) nad);
          }
--- 107,111 ----

              /* hand it to the app */
!             if ((plugin_error == 0) && (s->state < state_CLOSING))
                  _sx_event(s, event_PACKET, (void *) nad);
          }
