diff -ur jabberd-2.2.17.orig/tools/Makefile.am jabberd-2.2.17/tools/Makefile.am
--- jabberd-2.2.17.orig/tools/Makefile.am	2012-05-04 18:17:10.000000000 +0000
+++ jabberd-2.2.17/tools/Makefile.am	2013-07-23 17:39:51.154241563 +0000
@@ -8,6 +8,7 @@
 
 edit = sed \
 	-e 's,@sysconfdir\@,$(sysconfdir),g' \
+	-e 's,@libdir\@,$(libdir)/jabberd,g' \
 	-e 's,@VERSION\@,$(VERSION),g' \
 	-e 's,@bindir\@,$(bindir),g' \
 	-e 's,@libexecdir\@,$(libexecdir),g'
