diff -u -r1.27.2.5 util/nad.c
--- util/nad.c       30 Apr 2004 02:43:02 -0000      1.27.2.5
+++ util/nad.c       16 Sep 2004 17:10:32 -0000
@@ -726,7 +726,7 @@
     for(ns = nad->elems[elem].ns; ns >= 0; ns = nad->nss[ns].next)
     {
         /* never explicitly declare the implicit xml namespace */
-        if(nad->nss[ns].lprefix == strlen(uri_XML) && strncmp(uri_XML, nad->cdata + nad->nss[ns].iprefix, nad->nss[ns].lprefix) == 0)
+        if(nad->nss[ns].luri == strlen(uri_XML) && strncmp(uri_XML, nad->cdata + nad->nss[ns].iuri, nad->nss[ns].luri) == 0)
             continue;

         /* make space */ 
