diff -ur jabberd-2.0cvs.orig/configure.in jabberd-2.0cvs/configure.in
--- jabberd-2.0cvs.orig/configure.in	Thu Apr 14 20:27:07 2005
+++ jabberd-2.0cvs/configure.in	Sat Apr 16 11:25:10 2005
@@ -381,6 +381,22 @@
 fi
 
 
+dnl SQLite 3
+AC_ARG_ENABLE(sqlite, AC_HELP_STRING([--enable-sqlite], [enable SQLite 3 storage support (no)]),
+              want_sqlite=$enableval, want_sqlite=no)
+if test "x-$want_sqlite" = "x-yes" ; then
+    AC_CHECK_HEADERS(sqlite3.h)
+    if test "x-$ac_cv_header_sqlite3_h" = "x-yes" ; then
+        AC_CHECK_LIB(sqlite3, sqlite3_open)
+    fi
+    if test "x-$ac_cv_lib_sqlite3_sqlite3_open" != "x-yes" ; then
+        AC_MSG_ERROR([SQLite 3 library not found])
+    else
+        AC_DEFINE(STORAGE_SQLITE,1,[Define to 1 if you want to use SQLite 3 for storage.])
+    fi
+fi
+
+
 dnl berkeley db
 AC_ARG_ENABLE(db, AC_HELP_STRING([--enable-db], [enable Berkeley DB auth/reg/storage support (no)]),
               want_db=$enableval, want_db=no)
diff -ur jabberd-2.0cvs.orig/sm/Makefile.am jabberd-2.0cvs/sm/Makefile.am
--- jabberd-2.0cvs.orig/sm/Makefile.am	Thu Apr 14 20:19:26 2005
+++ jabberd-2.0cvs/sm/Makefile.am	Sat Apr 16 11:24:33 2005
@@ -20,6 +20,7 @@
              storage_mysql.c \
              storage_pgsql.c \
 	     storage_oracle.c \
+             storage_sqlite.c \
              mod_active.c \
              mod_announce.c \
              mod_deliver.c \
diff -ur jabberd-2.0cvs.orig/sm/storage.c jabberd-2.0cvs/sm/storage.c
--- jabberd-2.0cvs.orig/sm/storage.c	Thu Apr 14 20:27:07 2005
+++ jabberd-2.0cvs/sm/storage.c	Sat Apr 16 11:24:00 2005
@@ -48,6 +48,9 @@
 #ifdef STORAGE_ORACLE
 extern st_ret_t st_oracle_init(st_driver_t);
 #endif
+#ifdef STORAGE_SQLITE
+extern st_ret_t st_sqlite_init(st_driver_t);
+#endif
 
 static char *st_driver_names[] = {
 #ifdef STORAGE_DB
@@ -65,6 +68,9 @@
 #ifdef STORAGE_ORACLE
     "oracle",
 #endif
+#ifdef STORAGE_SQLITE
+    "sqlite",
+#endif
     NULL
 };
 
@@ -83,6 +89,9 @@
 #endif
 #ifdef STORAGE_ORACLE
     st_oracle_init,
+#endif
+#ifdef STORAGE_SQLITE
+    st_sqlite_init,
 #endif
     NULL
 };
