--- jabberd-2.0s6/tools/db-setup.mysql.orig	2005-03-13 12:59:32.477254560 +0100
+++ jabberd-2.0s6/tools/db-setup.mysql	2005-03-13 12:59:40.701004360 +0100
@@ -89,7 +89,9 @@
     `adr-pcode` TEXT,
     `adr-country` TEXT,
     `org-orgname` TEXT,
-    `org-orgunit` TEXT );
+    `org-orgunit` TEXT,
+    `photo-type` TEXT,
+    `photo-binval` TEXT );
 
 --
 -- Offline message queue
--- jabberd-2.0s6/tools/db-setup.pgsql~	2003-08-20 04:35:33.000000000 +0200
+++ jabberd-2.0s6/tools/db-setup.pgsql	2005-03-13 12:59:05.527351568 +0100
@@ -91,7 +91,9 @@
     "adr-pcode" text,
     "adr-country" text,
     "org-orgname" text,
-    "org-orgunit" text );
+    "org-orgunit" text,
+    "photo-type" text,
+    "photo-binval" text );
 
 --
 -- Offline message queue
--- jabberd-2.0s6/tools/db-setup.sqlite~	2005-03-13 12:55:06.911626672 +0100
+++ jabberd-2.0s6/tools/db-setup.sqlite	2005-03-13 12:58:31.658500416 +0100
@@ -87,7 +87,9 @@
     'adr-pcode' TEXT,
     'adr-country' TEXT,
     'org-orgname' TEXT,
-    'org-orgunit' TEXT );
+    'org-orgunit' TEXT,
+    'photo-type' TEXT,
+    'photo-binval' TEXT );
 
 --
 -- Offline message queue
--- jabberd-2.0s6/AVATARS.none	1970-01-01 01:00:00.000000000 +0100
+++ jabberd-2.0s6/AVATARS	2005-03-13 13:06:47.381139120 +0100
@@ -0,0 +1,7 @@
+To use avatars in vcard (when you use SQL)
+PLEASE ALTER TABLE AS FOLLOWING MYSQL EXAMPLE:
+
+ALTER TABLE `vcard` ADD `photo-type` TEXT, ADD `photo-binval` TEXT;
+
+Cheers.
+
