diff -durN jabberd-2.0b1.orig/etc/c2s.xml.dist.in jabberd-2.0b1/etc/c2s.xml.dist.in
--- jabberd-2.0b1.orig/etc/c2s.xml.dist.in	Fri Aug 29 12:31:38 2003
+++ jabberd-2.0b1/etc/c2s.xml.dist.in	Mon Sep 15 14:27:36 2003
@@ -6,7 +6,7 @@
   <!-- The process ID file. comment this out if you don't need to know
        to know the process ID from outside the process (eg for control
        scripts) -->
-  <pidfile>@localstatedir@/jabberd/pid/c2s.pid</pidfile>
+  <pidfile>@localstatedir@/run/jabberd/c2s.pid</pidfile>
 
   <!-- Router connection configuration -->
   <router>
@@ -34,7 +34,7 @@
 
     <!-- If logging to file, this is the filename of the logfile -->
     <!--
-    <file>@localstatedir@/jabberd/log/c2s.log</file>
+    <file>@localstatedir@/log/jabberd/c2s.log</file>
     -->
   </log>
 
@@ -177,7 +177,7 @@
   <!-- Authentication/registration database configuration -->
   <authreg>
     <!-- Backend module to use -->
-    <module>mysql</module>
+    <module>db</module>
 
     <!-- Registration configuration -->
     <register>
@@ -260,7 +260,7 @@
     <!-- Berkeley DB module configuration -->
     <db>
       <!-- Directory to store database files under -->
-      <path>@localstatedir@/jabberd/db</path>
+      <path>@localstatedir@/lib/jabberd/db</path>
       
       <!-- Synchronize the database to disk after each write. If you
            disable this, database accesses may be faster, but data may
diff -durN jabberd-2.0b1.orig/etc/jabberd.cfg.dist.in jabberd-2.0b1/etc/jabberd.cfg.dist.in
--- jabberd-2.0b1.orig/etc/jabberd.cfg.dist.in	Wed Aug 13 02:28:12 2003
+++ jabberd-2.0b1/etc/jabberd.cfg.dist.in	Mon Sep 15 14:27:36 2003
@@ -15,5 +15,10 @@
 @jabberd_router_bin@      @sysconfdir@/router.xml
 @jabberd_resolver_bin@    @sysconfdir@/resolver.xml
 @jabberd_sm_bin@          @sysconfdir@/sm.xml
-@jabberd_s2s_bin@         @sysconfdir@/s2s.xml
 @jabberd_c2s_bin@         @sysconfdir@/c2s.xml
+
+
+# After sm and c2s are configured to use a fully qualified domain name
+# and proper SRV records are set in DNS uncoment this to enable communication
+# with other Jabber servers
+#@jabberd_s2s_bin@         @sysconfdir@/s2s.xml
diff -durN jabberd-2.0b1.orig/etc/jabberd.cfg.dist.in.orig jabberd-2.0b1/etc/jabberd.cfg.dist.in.orig
--- jabberd-2.0b1.orig/etc/jabberd.cfg.dist.in.orig	Thu Jan  1 01:00:00 1970
+++ jabberd-2.0b1/etc/jabberd.cfg.dist.in.orig	Mon Sep 15 14:27:36 2003
@@ -0,0 +1,19 @@
+#
+# jabberd config file
+#
+#
+#   This file tells the jabberd wrapper what programs to launch,
+# and the config files to launch them with.  If the config file
+# is left out, then the system default will be used.
+#
+#   To run multiple Session Managers, just list them all seperatly
+# and provide the path to the appropriate config files.
+#
+# program   [ path to config file ]
+#
+
+@jabberd_router_bin@      @sysconfdir@/router.xml
+@jabberd_resolver_bin@    @sysconfdir@/resolver.xml
+@jabberd_sm_bin@          @sysconfdir@/sm.xml
+@jabberd_s2s_bin@         @sysconfdir@/s2s.xml
+@jabberd_c2s_bin@         @sysconfdir@/c2s.xml
diff -durN jabberd-2.0b1.orig/etc/jabberd.cfg.dist.in.rej jabberd-2.0b1/etc/jabberd.cfg.dist.in.rej
--- jabberd-2.0b1.orig/etc/jabberd.cfg.dist.in.rej	Thu Jan  1 01:00:00 1970
+++ jabberd-2.0b1/etc/jabberd.cfg.dist.in.rej	Mon Sep 15 14:27:36 2003
@@ -0,0 +1,18 @@
+***************
+*** 15,19 ****
+  router      @sysconfdir@/router.xml.dist.in
+  resolver    @sysconfdir@/resolver.xml.dist.in
+  sm          @sysconfdir@/sm.xml.dist.in
+- s2s         @sysconfdir@/s2s.xml.dist.in
+  c2s         @sysconfdir@/c2s.xml.dist.in
+--- 15,24 ----
+  router      @sysconfdir@/router.xml.dist.in
+  resolver    @sysconfdir@/resolver.xml.dist.in
+  sm          @sysconfdir@/sm.xml.dist.in
+  c2s         @sysconfdir@/c2s.xml.dist.in
++ 
++ 
++ # After sm and c2s are configured to use a fully qualified domain name
++ # and proper SRV records are set in DNS uncoment this to enable communication
++ # with other Jabber servers
++ #s2s         @sysconfdir@/s2s.xml.dist.in
diff -durN jabberd-2.0b1.orig/etc/resolver.xml.dist.in jabberd-2.0b1/etc/resolver.xml.dist.in
--- jabberd-2.0b1.orig/etc/resolver.xml.dist.in	Mon Aug 25 23:12:17 2003
+++ jabberd-2.0b1/etc/resolver.xml.dist.in	Mon Sep 15 14:27:36 2003
@@ -6,7 +6,7 @@
   <!-- The process ID file. comment this out if you don't need to know
        to know the process ID from outside the process (eg for control
        scripts) -->
-  <pidfile>@localstatedir@/jabberd/pid/resolver.pid</pidfile>
+  <pidfile>@localstatedir@/run/jabberd/resolver.pid</pidfile>
 
   <!-- Router connection configuration -->
   <router>
@@ -34,7 +34,7 @@
 
     <!-- If logging to file, this is the filename of the logfile -->
     <!--
-    <file>@localstatedir@/jabberd/log/resolver.log</file>
+    <file>@localstatedir@/log/jabberd/resolver.log</file>
     -->
   </log>
 
diff -durN jabberd-2.0b1.orig/etc/router.xml.dist.in jabberd-2.0b1/etc/router.xml.dist.in
--- jabberd-2.0b1.orig/etc/router.xml.dist.in	Tue Aug 26 03:28:45 2003
+++ jabberd-2.0b1/etc/router.xml.dist.in	Mon Sep 15 14:29:23 2003
@@ -6,7 +6,7 @@
   <!-- The process ID file. comment this out if you don't need to know
        to know the process ID from outside the process (eg for control
        scripts) -->
-  <pidfile>@localstatedir@/jabberd/pid/router.pid</pidfile>
+  <pidfile>@localstatedir@/run/jabberd/router.pid</pidfile>
 
   <!-- Log configuration - type is either "syslog" or "file" -->
   <log type='syslog'>
@@ -15,14 +15,14 @@
 
     <!-- If logging to file, this is the filename of the logfile -->
     <!--
-    <file>@localstatedir@/jabberd/log/router.log</file>
+    <file>@localstatedir@/log/jabberd/router.log</file>
     -->
   </log>
 
   <!-- Local network configuration -->
   <local>
     <!-- IP address to bind to (default: 0.0.0.0) -->
-    <ip>0.0.0.0</ip>
+    <ip>127.0.0.1</ip>
 
     <!-- Port to bind to (default: 5347) -->
     <port>5347</port>
@@ -123,6 +123,9 @@
     <!--
     <alias name='msn.domain.com' target='msn-linker'/>
     -->
+    <alias name='jud.localhost' target='jud-linker'/>
+    <alias name='muc.localhost' target='muc-linker'/>
+    <alias name='icq.localhost' target='icq-linker'/>
   </aliases>
 
   <!-- Access control information -->
diff -durN jabberd-2.0b1.orig/etc/s2s.xml.dist.in jabberd-2.0b1/etc/s2s.xml.dist.in
--- jabberd-2.0b1.orig/etc/s2s.xml.dist.in	Tue Aug 26 03:37:09 2003
+++ jabberd-2.0b1/etc/s2s.xml.dist.in	Mon Sep 15 14:27:36 2003
@@ -6,7 +6,7 @@
   <!-- The process ID file. comment this out if you don't need to know
        to know the process ID from outside the process (eg for control
        scripts) -->
-  <pidfile>@localstatedir@/jabberd/pid/s2s.pid</pidfile>
+  <pidfile>@localstatedir@/run/jabberd/s2s.pid</pidfile>
 
   <!-- Router connection configuration -->
   <router>
@@ -44,7 +44,7 @@
 
     <!-- if logging to file, this is the filename of the logfile -->
     <!--
-    <file>@localstatedir@/jabberd/log/s2s.log</file>
+    <file>@localstatedir@/log/jabberd/s2s.log</file>
     -->
   </log>
 
diff -durN jabberd-2.0b1.orig/etc/sm.xml.dist.in jabberd-2.0b1/etc/sm.xml.dist.in
--- jabberd-2.0b1.orig/etc/sm.xml.dist.in	Tue Sep  2 02:03:00 2003
+++ jabberd-2.0b1/etc/sm.xml.dist.in	Mon Sep 15 14:28:16 2003
@@ -9,7 +9,7 @@
   <!-- The process ID file. comment this out if you don't need to know
        to know the process ID from outside the process (eg for control
        scripts) -->
-  <pidfile>@localstatedir@/jabberd/pid/sm.pid</pidfile>
+  <pidfile>@localstatedir@/run/jabberd/sm.pid</pidfile>
 
   <!-- Router connection configuration -->
   <router>
@@ -37,14 +37,14 @@
 
     <!-- If logging to file, this is the filename of the logfile -->
     <!--
-    <file>@localstatedir@/jabberd/log/sm.log</file>
+    <file>@localstatedir@/log/jabberd/sm.log</file>
     -->
   </log>
 
   <!-- Storage database configuration -->
   <storage>
     <!-- By default, we use the MySQL driver for all storage -->
-    <driver>mysql</driver>
+    <driver>db</driver>
 
     <!-- Other storage drivers for specific purposes -->
     <!--
@@ -96,7 +96,7 @@
     <!-- Berkeley DB driver configuration -->
     <db>
       <!-- Directory to store database files under -->
-      <path>@localstatedir@/jabberd/db</path>
+      <path>@localstatedir@/lib/jabberd/db</path>
 
       <!-- Synchronize the database to disk after each write. If you
            disable this, database accesses may be faster, but data may
@@ -331,6 +331,8 @@
 
     </items>
 
+    <browse/>
+
   </discovery>
 
   <!-- User options -->
