*** util/util.h.orig	Mon May 31 23:31:06 2004
--- util/util.h	Mon Sep 27 14:40:53 2004
***************
*** 107,112 ****
--- 109,115 ----
  {
      int size;
      struct pfree *cleanup;
+     struct pfree *cleanup_tail;
      struct pheap *heap;
  #ifdef POOL_DEBUG
      char name[8], zone[32];
*** util/pool.c.orig	Mon May 31 23:31:06 2004
--- util/pool.c	Mon Sep 27 14:56:54 2004
***************
*** 85,97 ****
      if(p->cleanup == NULL)
      {
          p->cleanup = pf;
          return;
      }
  
!     /* fast forward to end of list */
!     for(cur = p->cleanup; cur->next != NULL; cur = cur->next);
! 
      cur->next = pf;
  }
  
  /** create a cleanup tracker */
--- 85,98 ----
      if(p->cleanup == NULL)
      {
          p->cleanup = pf;
+         p->cleanup_tail = pf;
          return;
      }
  
!     /* append at end of list */
!     cur = p->cleanup_tail; 
      cur->next = pf;
+     p->cleanup_tail = pf;
  }
  
  /** create a cleanup tracker */
