*** scod/mech_plain.c.orig	Fri Sep 17 22:44:21 2004
--- scod/mech_plain.c	Fri Sep 17 22:47:13 2004
***************
*** 64,68 ****
      c++;
  
!     pass = (char *) malloc(sizeof(char) * (resplen - ((int) (c - resp))));
      strncpy(pass, c, (resplen - ((int) (c - resp))));
      pass[resplen - ((int) (c - resp))] = '\0';
--- 64,68 ----
      c++;
  
!     pass = (char *) malloc(sizeof(char) * (resplen - ((int) (c - resp)) + 1));
      strncpy(pass, c, (resplen - ((int) (c - resp))));
      pass[resplen - ((int) (c - resp))] = '\0';
