--- jabberd-2.1.22/etc/c2s.xml.dist.in.orig	2008-01-21 16:17:03.000000000 +0100
+++ jabberd-2.1.22/etc/c2s.xml.dist.in	2008-01-22 14:38:12.000000000 +0100
@@ -5,7 +5,7 @@
 
   <!-- The process ID file. Comment this out if you don't need to know
        the process ID from outside the process (eg for control scripts) -->
-  <pidfile>@localstatedir@/jabberd/pid/c2s.pid</pidfile>
+  <pidfile>@localstatedir@/run/jabber/c2s.pid</pidfile>
 
   <!-- Router connection configuration -->
   <router>
@@ -60,7 +60,7 @@
 
     <!-- If logging to file, this is the filename of the logfile -->
     <!--
-    <file>@localstatedir@/jabberd/log/c2s.log</file>
+    <file>@localstatedir@/log/jabberd/c2s.log</file>
     -->
   </log>
 
@@ -284,7 +284,7 @@
   <stats>
     <!-- file containing count of packets that went through -->
     <!--
-    <packet>@localstatedir@/jabberd/stats/c2s.packets</packet>
+    <packet>@localstatedir@/lib/jabberd/stats/c2s.packets</packet>
     -->
   </stats>
 
@@ -292,7 +292,7 @@
     <path>@pkglibdir@</path>
 
     <!-- Backend module to use -->
-    <module>mysql</module>
+    <module>db</module>
 
     <!-- Available authentication mechanisms -->
     <mechanisms>
@@ -391,7 +391,7 @@
     <!-- SQLite driver configuration -->
     <sqlite>
       <!-- Database name -->
-      <dbname>@localstatedir@/jabberd/db/sqlite.db</dbname>
+      <dbname>@localstatedir@/lib/jabberd/sqlite.db</dbname>
 
       <!-- Transacation support. If this is commented out, transactions
            will be disabled. This might make database accesses faster,
@@ -405,7 +405,7 @@
     <!-- Berkeley DB module configuration -->
     <db>
       <!-- Directory to store database files under -->
-      <path>@localstatedir@/jabberd/db</path>
+      <path>@localstatedir@/lib/jabberd/db</path>
 
       <!-- Synchronize the database to disk after each write. If you
            disable this, database accesses may be faster, but data may
--- jabberd-2.1.22/etc/jabberd.cfg.dist.in.orig	2008-01-22 14:17:57.000000000 +0100
+++ jabberd-2.1.22/etc/jabberd.cfg.dist.in	2008-01-22 14:17:57.000000000 +0100
@@ -15,5 +15,10 @@
 @jabberd_router_bin@      @sysconfdir@/router.xml
 @jabberd_resolver_bin@    @sysconfdir@/resolver.xml
 @jabberd_sm_bin@          @sysconfdir@/sm.xml
-@jabberd_s2s_bin@         @sysconfdir@/s2s.xml
 @jabberd_c2s_bin@         @sysconfdir@/c2s.xml
+
+
+# After sm and c2s are configured to use a fully qualified domain name
+# and proper SRV records are set in DNS uncoment this to enable communication
+# with other Jabber servers
+#@jabberd_s2s_bin@         @sysconfdir@/s2s.xml
--- jabberd-2.1.22/etc/router.xml.dist.in.orig	2008-01-21 16:17:03.000000000 +0100
+++ jabberd-2.1.22/etc/router.xml.dist.in	2008-01-22 14:33:34.000000000 +0100
@@ -5,7 +5,7 @@
 
   <!-- The process ID file. Comment this out if you don't need to know
        the process ID from outside the process (eg for control scripts) -->
-  <pidfile>@localstatedir@/jabberd/pid/router.pid</pidfile>
+  <pidfile>@localstatedir@/run/jabber/router.pid</pidfile>
 
   <!-- Log configuration - type is "syslog", "file" or "stdout" -->
   <log type='syslog'>
@@ -18,14 +18,14 @@
 
     <!-- If logging to file, this is the filename of the logfile -->
     <!--
-    <file>@localstatedir@/jabberd/log/router.log</file>
+    <file>@localstatedir@/log/jabberd/router.log</file>
     -->
   </log>
 
   <!-- Local network configuration -->
   <local>
     <!-- IP address to bind to (default: 0.0.0.0) -->
-    <ip>0.0.0.0</ip>
+    <ip>127.0.0.1</ip>
 
     <!-- Port to bind to (default: 5347) -->
     <port>5347</port>
@@ -152,6 +152,9 @@
     <!--
     <alias name='conference.domain.com' target='muclinker'/>
     -->
+    <alias name='jud.localhost' target='jud-linker'/>
+    <alias name='muc.localhost' target='muc-linker'/>
+    <alias name='icq.localhost' target='icq-linker'/>
   </aliases>
 
   <!-- Access control information -->
--- jabberd-2.1.22/etc/s2s.xml.dist.in.orig	2008-01-21 16:17:03.000000000 +0100
+++ jabberd-2.1.22/etc/s2s.xml.dist.in	2008-01-22 14:33:51.000000000 +0100
@@ -5,7 +5,7 @@
 
   <!-- The process ID file. Comment this out if you don't need to know
        the process ID from outside the process (eg for control scripts) -->
-  <pidfile>@localstatedir@/jabberd/pid/s2s.pid</pidfile>
+  <pidfile>@localstatedir@/run/jabber/s2s.pid</pidfile>
 
   <!-- Router connection configuration -->
   <router>
@@ -70,7 +70,7 @@
 
     <!-- if logging to file, this is the filename of the logfile -->
     <!--
-    <file>@localstatedir@/jabberd/log/s2s.log</file>
+    <file>@localstatedir@/log/jabberd/s2s.log</file>
     -->
   </log>
 
@@ -202,7 +202,7 @@
   <stats>
     <!-- file containing count of packets that went through -->
     <!--
-    <packet>@localstatedir@/jabberd/stats/s2s.packets</packet>
+    <packet>@localstatedir@/lib/jabberd/stats/s2s.packets</packet>
     -->
   </stats>
 
--- jabberd-2.1.22/etc/sm.xml.dist.in.orig	2008-01-21 16:17:03.000000000 +0100
+++ jabberd-2.1.22/etc/sm.xml.dist.in	2008-01-22 14:37:53.000000000 +0100
@@ -8,7 +8,7 @@
 
   <!-- The process ID file. Comment this out if you don't need to know
        the process ID from outside the process (eg for control scripts) -->
-  <pidfile>@localstatedir@/jabberd/pid/sm.pid</pidfile>
+  <pidfile>@localstatedir@/run/jabber/sm.pid</pidfile>
 
   <!-- Router connection configuration -->
   <router>
@@ -63,7 +63,7 @@
 
     <!-- If logging to file, this is the filename of the logfile -->
     <!--
-    <file>@localstatedir@/jabberd/log/sm.log</file>
+    <file>@localstatedir@/log/jabberd/sm.log</file>
     -->
   </log>
 
@@ -73,7 +73,7 @@
     <path>@pkglibdir@</path>
 
     <!-- By default, we use the MySQL driver for all storage -->
-    <driver>mysql</driver>
+    <driver>db</driver>
 
     <!-- Its also possible to explicitly list alternate drivers for
          specific data types. -->
@@ -152,7 +152,7 @@
     <!-- Berkeley DB driver configuration -->
     <db>
       <!-- Directory to store database files under -->
-      <path>@localstatedir@/jabberd/db</path>
+      <path>@localstatedir@/lib/jabberd/db</path>
 
       <!-- Synchronize the database to disk after each write. If you
            disable this, database accesses may be faster, but data may
@@ -177,7 +177,7 @@
     <!-- SQLite driver configuration -->
     <sqlite>
       <!-- Database name -->
-      <dbname>@localstatedir@/jabberd/db/sqlite.db</dbname>
+      <dbname>@localstatedir@/lib/jabberd/sqlite.db</dbname>
 
       <!-- Transacation support. If this is commented out, transactions
            will be disabled. This might make database accesses faster,
@@ -532,6 +532,8 @@
     </serverinfo>
     -->
 
+    <browse/>
+
   </discovery>
 
   <!-- User options -->
