--- jabberd-2.0s6/etc/sm.xml.dist.in~	2005-01-07 10:24:47.628651464 +0100
+++ jabberd-2.0s6/etc/sm.xml.dist.in	2005-01-07 10:33:17.303169160 +0100
@@ -65,7 +65,8 @@
 
   <!-- Storage database configuration -->
   <storage>
-    <!-- By default, we use the MySQL driver for all storage -->
+    <!-- By default, we use the MySQL driver for all storage 
+         You can type here: db, mysql, pgsql, sqlite -->
     <driver>db</driver>
 
     <!-- Its also possible to explicitly list alternate drivers for
@@ -127,6 +128,17 @@
            be lost if jabberd crashes. -->
       <sync/>
     </db>
+    
+     <sqlite>
+      <!-- Directory and filename of database -->
+      <dbname>@localstatedir@/lib/jabberd/jabberd2.db</dbname>
+
+      <!-- Transacation support. If this is commented out, transactions
+           will be disabled. This might make database accesses faster,
+           but data may be lost if jabberd crashes. -->
+      <transactions/>
+    </sqlite>
+
   </storage>
   
   <!-- Access control information -->
