--- jabberd-2.0s8/etc/sm.xml.dist.in.wiget	2005-04-25 21:10:23.000000000 +0200
+++ jabberd-2.0s8/etc/sm.xml.dist.in	2005-04-25 21:12:29.000000000 +0200
@@ -129,6 +129,17 @@
       <sync/>
     </db>
 
+	<!-- SQLite driver configuration -->
+    <sqlite>
+      <!-- Directory and filename of database -->
+      <dbname>@localstatedir@/lib/jabberd/jabberd2.db</dbname>
+
+      <!-- Transacation support. If this is commented out, transactions
+           will be disabled. This might make database accesses faster,
+           but data may be lost if jabberd crashes. -->
+      <transactions/>
+    </sqlite>
+
     <!-- Oracle driver configuration -->
     <oracle>
       <!-- Database server host and port. -->
@@ -387,6 +398,17 @@
     <auto-create/>
     -->
 
+  <!-- Offline patches -->
+    <!-- By default, no quota or dropping messages or subscriptions 
+	 are disabled. -->
+    <!--
+    <offline>
+      <userquota>500</userquota>
+      <dropmessages/>
+      <dropsubscriptions/>
+    </offline>
+    -->
+
     <!-- Templates. If defined, the contents of these files will be
          stored in the users data store when they are created. -->
     <template>
